Add initial docs and example implementation for UPI bare-metal

abhinavdahiya opened the pull request:
* Adds an example implementation for UPI Bare Metal in `upi/metal` using Packet.net and AWS.
* Adds docs on requirements and how-to on using OpenShift Installer to create a Bare Metal cluster.

abhinavdahiya commented:
/cc @kalexand-rh for tracking docs

DanyC97 commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+## Network Topology Requirements
+
+OpenShift 4.x requires all nodes to have internet access to pull images for platform containers and phone home with telemetry data.
```
nit: `home` is ?  maybe would be better to reword it such that the reason and the location is understood:

`phone home` -> `stream the telemetry data back to RedHat <telemetry central location>`

DanyC97 commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+1. Load balancer targeting the control plane machines
+
+    TCP load balancer for ports 6443 and 22623. The targets will be the control plane machines. Port 6443 must be accessible to both clients external to the cluster and nodes within the cluster. Port 22623 must be accessible to nodes within the cluster.
```
nit: will be nice (especially for network admins who will ask this question) to add a note or cross link the usage/ consumer of port `22623`. Potentially same could be done for `6443` although is known from K8s/ OpenShift docs

DanyC97 commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+    For each control plane machine, OpenShift 4.x requires DNS records `etcd-$idx.$cluster_name.$base_domain` to point to `$idx`'th control plane machine. The DNS record must resolve to an unicast IPV4 address for the control plane machine and the records must be resolvable from all the nodes in the cluster.
+
+    For each control plane machine, OpenShift 4.x also requires a SRV DNS record for etcd server on that machine with priority `0`, weight `10` and port `2380`. For 3 control plane cluster, the records look like:
+
```
nice one, thank you for adding this paragraph! 👍

DanyC97 commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+# Install: User Provided Infrastructure
+
+The steps for performing a UPI-based install are outlined here. Several [Terraform][terraform-templates] templates are provided to help model your own.
```
@abhinavdahiya you think is sensible to extend the above sentence to suggest that the terraform templates are just an example however the user/ operator is free to model their own infrastructure using whatever CM/ orchestration tool they want ? 

i believe by doing so nobody (external to RH) will get confuse, especially with this [info](https://github.com/openshift/openshift-ansible/pull/11338) and so the RH story will flow

my 0.02$

abhinavdahiya commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+    TCP load balancer for ports 443 and 80. The target will be the machines running the ingress router [todo-link-to-openshift-router]. Both the ports must be accessible to both clients external to the cluster and nodes within the cluster.
+
+    NOTE: The default configuration for Cluster Ingress Operator [todo-link-to-configuring-router-using-operator] deploys the ingress router to `worker` nodes in the cluster. A working configuration for the ingress router is required for an OpenShift 4.x cluster.
```
Nice catch! 

added `The administrator needs to configure the ingress after the control plane has been bootstrapped` to clarify.

abhinavdahiya commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+## Compute
+
+The smallest OpenShift 4.x clusters require, 1 bootstrap machine, 3 control plane machines and at least 1 worker machine.
```
we use machine for hardware in installer at many places. so `machines` keep consistency. Are you concerned in the in `bare metal` context machines would not be correct term for hardware host.

kalexand-rh replied:
Our style guide says to not use "machine" to mean hardware if there's another term.

wking replied:
> Our style guide says to not use "machine" to mean hardware if there's another term.

"Machine" is also what the machine API uses for this, so I'm in favor of "machine" for consistency.  Does the style guide motivate "host" vs. "machine"?  I'm also fine with "node" for control-plane and compute entries, but the bootstrap machine is not a node in the Kubernetes sense, so we don't have that out there.

abhinavdahiya commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+Network connectivity between machines is required on various ports for platform components to work.
+
+1. Etcd
```
I'm not sure I understand which files?

kalexand-rh replied:
I thought you were trying to link to files deeper in the repo to explain etcd.

abhinavdahiya commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,107 @@
+
+### Connectivity between Machines
+
+Network connectivity between machines is required on various ports for platform components to work.
```
The networking firewalls have ingress and egress rules that together need to be configured to achieve the connectivity. That is going to get a little verbose and difficult to manage in this doc. that is why decided to go with small sentences that provide gist... ?

kalexand-rh replied:
That's fair! After the final data is available, I might try to put them into a table in the doc repo.

abhinavdahiya commented:
@wking @sdodson  this is ready for review. PTAL

sdodson commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,305 @@
+
+## Getting OS related assets for machines
+
+TODO RHEL CoreOS does not have assets for bare metal.
```
Is this to include where to put the *.ign files which are referenced later in pxe config or should we go ahead and mention hosting those files in the previous section?

abhinavdahiya replied:
this is how to get RHCOS iso / os-raw-image ..

sdodson replied:
Ok, I think in the previous section we should mention the need to store *.ign files somewhere that's accessible to the nodes.

abhinavdahiya replied:
https://github.com/openshift/installer/pull/1416/files/972d6a56d436fba77a4d7b70d653ee614b32472e#diff-1a64e7030769df4d25b1c09ddb8f62deR199 
`coreos-installer` is how RHCOS boots on metal. and it has `coreos.inst.ignition_url` kernel parameter

abhinavdahiya commented:
Using the terraform example:
```
```

[bootstrap-complete]
```console
$ ../../bin/openshift-install --dir dev-metal user-provided-infrastructure bootstrap-complete --log-level=debug
DEBUG OpenShift Installer unreleased-master-582-ga726aeb61c29467064860afdcbbfc4b1cf804bfe
DEBUG Built from commit a726aeb61c29467064860afdcbbfc4b1cf804bfe
INFO Waiting up to 30m0s for the Kubernetes API at https://api.adahiya-0.devcluster.openshift.com:6443...
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.109.237:6443: connect: connection refused
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.109.237:6443: i/o timeout
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.68.117:6443: connect: connection refused
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.68.117:6443: connect: connection refused
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.68.117:6443: i/o timeout
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.109.237:6443: connect: connection refused
DEBUG Still waiting for the Kubernetes API: the server could not find the requested resource
DEBUG Still waiting for the Kubernetes API: the server could not find the requested resource
DEBUG Still waiting for the Kubernetes API: the server could not find the requested resource
DEBUG Still waiting for the Kubernetes API: the server could not find the requested resource
DEBUG Still waiting for the Kubernetes API: the server could not find the requested resource
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: dial tcp 147.75.109.237:6443: connect: connection refused
DEBUG Still waiting for the Kubernetes API: Get https://api.adahiya-0.devcluster.openshift.com:6443/version?timeout=32s: net/http: TLS handshake timeout
INFO API v1.12.4+41bf871 up
INFO Waiting up to 30m0s for the bootstrap-complete event...
DEBUG added kube-scheduler.158d84e5fe5b939b: adahiya-0-23456-bootstrap_608083b0-4aab-11e9-9f67-0cc47a540d02 became leader
DEBUG added kube-controller-manager.158d84e6cd963cf3: adahiya-0-23456-bootstrap_63204c56-4aab-11e9-8292-0cc47a540d02 became leader
DEBUG added bootstrap-success: Required control plane pods have been created
DEBUG added openshift-master-controllers.158d8519469a38dc: controller-manager-q8w5v became leader
DEBUG added bootstrap-complete: cluster bootstrapping has completed
INFO It is now safe to remove the bootstrap resources
```

[2 node cluster]
```console
oc --config dev-metal/auth/kubeconfig get nodes -owide
NAME                       STATUS   ROLES    AGE     VERSION              INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                                                   KERNEL-VERSION         CONTAINER-RUNTIME
adahiya-0-23456-master-0   Ready    master   7m54s   v1.12.4+d14915559e   147.75.68.117    <none>        Red Hat Enterprise Linux CoreOS 410.8.20190319.1 (Ootpa)   4.18.0-80.el8.x86_64   cri-o://1.12.8-1.rhaos4.0.git35a0cc1.el7
adahiya-0-23456-worker-0   Ready    worker   7m57s   v1.12.4+d14915559e   147.75.109.199   <none>        Red Hat Enterprise Linux CoreOS 410.8.20190319.1 (Ootpa)   4.18.0-80.el8.x86_64   cri-o://1.12.8-1.rhaos4.0.git35a0cc1.el7
```

[dettach bootstrap node from api DNS]
```console
$ source packet/api-keys && AWS_PROFILE=openshift-dev terraform apply -var=bootstrap_dns="false"
matchbox_profile.bootstrap: Refreshing state... (ID: adahiya-0-23456-bootstrap)
matchbox_profile.worker: Refreshing state... (ID: adahiya-0-23456-worker)
matchbox_profile.master: Refreshing state... (ID: adahiya-0-23456-master)
matchbox_group.worker: Refreshing state... (ID: adahiya-0-23456-worker)
matchbox_group.master: Refreshing state... (ID: adahiya-0-23456-master)
packet_device.workers: Refreshing state... (ID: 044f93ca-6b97-41b5-966c-b259c1d32b60)
packet_device.masters: Refreshing state... (ID: dad60c3e-c760-4e3f-893a-219b671bdb1a)
matchbox_group.bootstrap: Refreshing state... (ID: adahiya-0-23456-bootstrap)
packet_device.bootstrap: Refreshing state... (ID: e3d4ba00-c27b-460d-8b97-cb68803dbde9)
data.template_file.worker_ips: Refreshing state...
data.template_file.master_ips: Refreshing state...
data.aws_route53_zone.public: Refreshing state...
aws_route53_record.apps: Refreshing state... (ID: Z3URY6TWQ91KVV_*.apps.adahiya-0.devcluster.openshift.com_A)
aws_route53_record.ctrlp: Refreshing state... (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A)
aws_route53_record.etcd_a_nodes: Refreshing state... (ID: Z3URY6TWQ91KVV_etcd-0.adahiya-0.devcluster.openshift.com_A)
aws_route53_record.etcd_cluster: Refreshing state... (ID: Z3URY6TWQ91KVV__etcd-server-ssl._tcp.adahiya-0.devcluster.openshift.com_SRV)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_route53_record.ctrlp
      records.#:          "2" => "1"
      records.3528999564: "147.75.68.117" => "147.75.68.117"
      records.453449262:  "147.75.109.237" => ""


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_route53_record.ctrlp: Modifying... (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A)
  records.#:          "2" => "1"
  records.3528999564: "147.75.68.117" => "147.75.68.117"
  records.453449262:  "147.75.109.237" => ""
aws_route53_record.ctrlp: Still modifying... (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A, 10s elapsed)
aws_route53_record.ctrlp: Still modifying... (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A, 20s elapsed)
aws_route53_record.ctrlp: Still modifying... (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A, 30s elapsed)
aws_route53_record.ctrlp: Modifications complete after 32s (ID: Z3URY6TWQ91KVV_api.adahiya-0.devcluster.openshift.com_A)

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

bootstrap_ip = 147.75.109.237
master_ips = [
    147.75.68.117
]
sos_consoles = {
  adahiya-0-23456-bootstrap = ssh e3d4ba00-c27b-460d-8b97-cb68803dbde9@sos.sjc1.packet.net
  adahiya-0-23456-master-0 = ssh dad60c3e-c760-4e3f-893a-219b671bdb1a@sos.sjc1.packet.net
  adahiya-0-23456-worker-0 = ssh 044f93ca-6b97-41b5-966c-b259c1d32b60@sos.sjc1.packet.net
}
```

Known issue the nodes come up with non-resolvable hostnames
```console
oc --config dev-metal/auth/kubeconfig logs cluster-version-operator-764c6f86d8-pvgnr -n openshift-cluster-version
Error from server: Get https://adahiya-0-23456-master-0:10250/containerLogs/openshift-cluster-version/cluster-version-operator-764c6f86d8-pvgnr/cluster-version-operator: dial tcp: lookup adahiya-0-23456-master-0: no such host
```

cluster deployment stuck at
```console
$ oc --config dev-metal/auth/kubeconfig get clusterversion version -oyaml
apiVersion: config.openshift.io/v1
kind: ClusterVersion
metadata:
  creationTimestamp: 2019-03-20T00:59:41Z
  generation: 1
  name: version
  resourceVersion: "4734"
  selfLink: /apis/config.openshift.io/v1/clusterversions/version
  uid: 7138f8a9-4aab-11e9-9b87-0cc47a540d02
spec:
  channel: stable-4.0
  clusterID: 640f6438-e5f3-4a5e-9a6e-0ee9c1137ff6
  upstream: https://api.openshift.com/api/upgrades_info/v1/graph
status:
  availableUpdates: null
  conditions:
  - lastTransitionTime: 2019-03-20T00:59:47Z
    status: "False"
    type: Available
  - lastTransitionTime: 2019-03-20T00:59:47Z
    status: "False"
    type: Failing
  - lastTransitionTime: 2019-03-20T00:59:47Z
    message: 'Working towards 4.0.0-0.alpha-2019-03-19-200331: 66% complete'
    status: "True"
    type: Progressing
  - lastTransitionTime: 2019-03-20T00:59:47Z
    message: 'Unable to retrieve available updates: unknown version 4.0.0-0.alpha-2019-03-19-200331'
    reason: RemoteFailed
    status: "False"
    type: RetrievedUpdates
  desired:
    image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:4c8435500836dfe9e64692c68c68da761f5f539054ff5300368cfd12246a9704
    version: 4.0.0-0.alpha-2019-03-19-200331
  history:
  - completionTime: null
    image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:4c8435500836dfe9e64692c68c68da761f5f539054ff5300368cfd12246a9704
    startedTime: 2019-03-20T00:59:47Z
    state: Partial
    version: 4.0.0-0.alpha-2019-03-19-200331
  observedGeneration: 1
  versionHash: sp24qk8ln3Q=
$ oc --config dev-metal/auth/kubeconfig get co
NAME                                  VERSION                           AVAILABLE   PROGRESSING   FAILING   SINCE
cluster-autoscaler                                                      True        False         True      2s
dns                                   4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
kube-apiserver                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     9m26s
kube-controller-manager               4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
kube-scheduler                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
machine-config                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     12m
network                               4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
openshift-apiserver                   4.0.0-0.alpha-2019-03-19-200331   True        False         False     8m26s
openshift-cloud-credential-operator   4.0.0-0.alpha-2019-03-19-200331   True        False         False     12m
openshift-controller-manager          4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
operator-lifecycle-manager            4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
service-ca                            4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
```

smarterclayton commented on docs/user/metal/install_upi.md:
````diff
@@ -0,0 +1,305 @@
+The administrators can use the `user-provided-infrastructure bootstrap-complete` target of the OpenShift Installer to monitor cluster bootstrapping. The command succeeds when it notices `bootstrap-complete` event from Kubernetes APIServer. This event is generated by the bootstrap machine after the Kubernetes APIServer has been bootstrapped on the control plane machines. For example,
+
+```console
+$ openshift-install --dir test-bare-metal user-provided-infrastructure bootstrap-complete
````
That's a helluva command, length wise.

smarterclayton commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,305 @@
+
+## Example Bare-Metal UPI deployment
+
+Terraform [templates][upi-metal-example] provides and example of using OpenShift Installer to create an UPI Bare Metal OpenShift cluster on Packet.net
```
> ... provides an ...

smarterclayton replied:
and

> create a UPI Bare

Would probably suggest we switch `Bare Metal` to `bare-metal` everywhere.

kalexand-rh replied:
+1. Use "bare-metal" as an adjective.

sdodson commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,305 @@
+
+The OpenShift Installer provides administrators various assets that are required to create an OpenShift cluster, namely:
+
+* Ignition configs: The OpenShift Installer provides Ignition configs that should be used to configure the RHCOS based boostrap and control plane machines using `bootstrap.ign`  and `master.ign` respectively. The OpenShift Installer also provides `worker.ign` that can be used to configure the RHCOS based `worker` machines, but also can be used as source for configuring RHEL based machines [todo-l ...
```
s/boostrap/bootstrap/

abhinavdahiya commented:
Known issue the nodes come up with non-resolvable hostnames

> 
> ```
> oc --config dev-metal/auth/kubeconfig logs cluster-version-operator-764c6f86d8-pvgnr -n openshift-cluster-version
> Error from server: Get https://adahiya-0-23456-master-0:10250/containerLogs/openshift-cluster-version/cluster-version-operator-764c6f86d8-pvgnr/cluster-version-operator: dial tcp: lookup adahiya-0-23456-master-0: no such host
> ```

Fixed 
- by adding hostnames that are globally resolvable using AWS route 53 https://github.com/openshift/installer/pull/1416/commits/10e990c7858b8996651be7c330104ad32c0b82b1
- by adding section that adds commands to approve all pending CSRs for kubelet server certificates.

oc logs now works:
```
oc --config dev-metal/auth/kubeconfig logs ingress-operator-7db46fb676-gz2th -n openshift-ingress-operator
2019-03-20T16:22:29.114Z        INFO    operator        log/log.go:26   started zapr logger
2019-03-20T16:22:29.272Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:121    Starting EventSource    {"controller": "operator-controller", "source": "kind source: /, Kind="}
2019-03-20T16:22:29.303Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:121    Starting EventSource    {"controller": "operator-controller", "source": "informer source: 0xc420560bd0"}
2019-03-20T16:22:29.303Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:121    Starting EventSource    {"controller": "operator-controller", "source": "informer source: 0xc420560c60"}
2019-03-20T16:22:29.303Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:121    Starting EventSource    {"controller": "certificate-controller", "source": "kind source: /, Kind="}
2019-03-20T16:22:29.303Z        INFO    operator.init   operator/operator.go:198        waiting for cache to sync
2019-03-20T16:22:29.317Z        INFO    operator.init   operator/operator.go:242        created default ingresscontroller       {"namespace": "openshift-ingress-operator", "name": "default"}
2019-03-20T16:22:29.404Z        INFO    operator.init   operator/operator.go:202        cache synced
2019-03-20T16:22:29.504Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:134    Starting Controller     {"controller": "operator-controller"}
2019-03-20T16:22:29.504Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:134    Starting Controller     {"controller": "certificate-controller"}
2019-03-20T16:22:29.604Z        INFO    operator.init.kubebuilder.controller    controller/controller.go:154    Starting workers        {"controller": "certificate-controller", "worker count": 1}
```

Moving ingress to HostNetwork

Patching ingress-controller is not updating ingress to HostNetwork

```console
$ oc --config dev-metal/auth/kubeconfig patch ingresscontrollers.operator.openshift.io default --type merge --patch '{"spec":{"endpointPublishingStrategy": {"type": "HostNetwork"}}}' -n openshift-ingress-operator
ingresscontroller.operator.openshift.io/default patched
## wait for few minutes..
$ oc --config dev-metal/auth/kubeconfig get ingresscontrollers.operator.openshift.io default -n openshift-ingress-operator -ojson | jq '.status'
{
  "availableReplicas": 1,
  "domain": "apps.adahiya-0.devcluster.openshift.com",
  "endpointPublishingStrategy": {
    "type": "Private"
  },
  "selector": "app=router,router=router-default"
}
```

> cluster deployment stuck at
> 
> ```
> $ oc --config dev-metal/auth/kubeconfig get clusterversion version -oyaml
> apiVersion: config.openshift.io/v1
> kind: ClusterVersion
> metadata:
>   creationTimestamp: 2019-03-20T00:59:41Z
>   generation: 1
>   name: version
>   resourceVersion: "4734"
>   selfLink: /apis/config.openshift.io/v1/clusterversions/version
>   uid: 7138f8a9-4aab-11e9-9b87-0cc47a540d02
> spec:
>   channel: stable-4.0
>   clusterID: 640f6438-e5f3-4a5e-9a6e-0ee9c1137ff6
>   upstream: https://api.openshift.com/api/upgrades_info/v1/graph
> status:
>   availableUpdates: null
>   conditions:
>   - lastTransitionTime: 2019-03-20T00:59:47Z
>     status: "False"
>     type: Available
>   - lastTransitionTime: 2019-03-20T00:59:47Z
>     status: "False"
>     type: Failing
>   - lastTransitionTime: 2019-03-20T00:59:47Z
>     message: 'Working towards 4.0.0-0.alpha-2019-03-19-200331: 66% complete'
>     status: "True"
>     type: Progressing
>   - lastTransitionTime: 2019-03-20T00:59:47Z
>     message: 'Unable to retrieve available updates: unknown version 4.0.0-0.alpha-2019-03-19-200331'
>     reason: RemoteFailed
>     status: "False"
>     type: RetrievedUpdates
>   desired:
>     image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:4c8435500836dfe9e64692c68c68da761f5f539054ff5300368cfd12246a9704
>     version: 4.0.0-0.alpha-2019-03-19-200331
>   history:
>   - completionTime: null
>     image: registry.svc.ci.openshift.org/openshift/origin-release@sha256:4c8435500836dfe9e64692c68c68da761f5f539054ff5300368cfd12246a9704
>     startedTime: 2019-03-20T00:59:47Z
>     state: Partial
>     version: 4.0.0-0.alpha-2019-03-19-200331
>   observedGeneration: 1
>   versionHash: sp24qk8ln3Q=
> $ oc --config dev-metal/auth/kubeconfig get co
> NAME                                  VERSION                           AVAILABLE   PROGRESSING   FAILING   SINCE
> cluster-autoscaler                                                      True        False         True      2s
> dns                                   4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
> kube-apiserver                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     9m26s
> kube-controller-manager               4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
> kube-scheduler                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
> machine-config                        4.0.0-0.alpha-2019-03-19-200331   True        False         False     12m
> network                               4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
> openshift-apiserver                   4.0.0-0.alpha-2019-03-19-200331   True        False         False     8m26s
> openshift-cloud-credential-operator   4.0.0-0.alpha-2019-03-19-200331   True        False         False     12m
> openshift-controller-manager          4.0.0-0.alpha-2019-03-19-200331   True        False         False     11m
> operator-lifecycle-manager            4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
> service-ca                            4.0.0-0.alpha-2019-03-19-200331   True        False         False     13m
> ```


After manually editing the router to HostNetwork we seems to be making progress..
```console
oc --config dev-metal/auth/kubeconfig get co
NAME                                  VERSION                           AVAILABLE   PROGRESSING   FAILING   SINCE
authentication                        4.0.0-0.alpha-2019-03-20-094557   True        False         False     156m
cluster-autoscaler                                                      True        False         True      9s
console                               4.0.0-0.alpha-2019-03-20-094557   True        False         False     31m
dns                                   4.0.0-0.alpha-2019-03-20-094557   True        False         False     161m
image-registry                                                          False       True          False     158m
ingress                               4.0.0-0.alpha-2019-03-20-094557   True        False         False     31m
kube-apiserver                        4.0.0-0.alpha-2019-03-20-094557   True        False         False     157m
kube-controller-manager               4.0.0-0.alpha-2019-03-20-094557   True        False         False     155m
kube-scheduler                        4.0.0-0.alpha-2019-03-20-094557   True        False         False     155m
machine-config                        4.0.0-0.alpha-2019-03-20-094557   False       True          True      161m
marketplace-operator                  4.0.0-0.alpha-2019-03-20-094557   True        False         False     157m
monitoring                            4.0.0-0.alpha-2019-03-20-094557   True        False         False     39m
network                               4.0.0-0.alpha-2019-03-20-094557   True        False         False     162m
node-tuning                           4.0.0-0.alpha-2019-03-20-094557   True        False         False     157m
openshift-apiserver                   4.0.0-0.alpha-2019-03-20-094557   True        False         False     158m
openshift-cloud-credential-operator   4.0.0-0.alpha-2019-03-20-094557   True        False         False     161m
openshift-controller-manager          4.0.0-0.alpha-2019-03-20-094557   True        False         False     155m
openshift-samples                     4.0.0-0.alpha-2019-03-20-094557   True        False         False     157m
operator-lifecycle-manager            4.0.0-0.alpha-2019-03-20-094557   True        False         False     161m
service-ca                            4.0.0-0.alpha-2019-03-20-094557   True        False         False     156m
service-catalog-apiserver             4.0.0-0.alpha-2019-03-20-094557   True        False         False     158m
service-catalog-controller-manager    4.0.0-0.alpha-2019-03-20-094557   True        False         False     155m
storage                                                                 True        False         False     158m
```

only registry and machine-config failing
```console
$ oc --config dev-metal/auth/kubeconfig get co -ojson | jq '.items[] | select(.status.conditions[] | .type == "Available" and .status != "True") | [.metadata.name, .status.conditions]'
[
  "image-registry",
  [
    {
      "lastTransitionTime": "2019-03-20T16:22:33Z",
      "message": "Deployment does not exist",
      "status": "False",
      "type": "Available"
    },
    {
      "lastTransitionTime": "2019-03-20T16:22:33Z",
      "message": "Unable to apply resources: unable to sync storage configuration: storage backend not configured",
      "status": "True",
      "type": "Progressing"
    },
    {
      "lastTransitionTime": "2019-03-20T16:22:33Z",
      "status": "False",
      "type": "Failing"
    }
  ]
]
[
  "machine-config",
  [
    {
      "lastTransitionTime": "2019-03-20T16:19:16Z",
      "message": "Cluster not available for 4.0.0-0.alpha-2019-03-20-094557",
      "status": "False",
      "type": "Available"
    },
    {
      "lastTransitionTime": "2019-03-20T16:19:16Z",
      "message": "Cluster is bootstrapping 4.0.0-0.alpha-2019-03-20-094557",
      "status": "True",
      "type": "Progressing"
    },
    {
      "lastTransitionTime": "2019-03-20T16:19:51Z",
      "message": "Failed to resync 4.0.0-0.alpha-2019-03-20-094557 because: error pool master is not ready. status: (total: 1, updated: 0, unavailable: 0)",
      "reason": "error pool master is not ready. status: (total: 1, updated: 0, unavailable: 0)",
      "status": "True",
      "type": "Failing"
    }
  ]
]
```

registry failure is concerning. need to figure out how to tell registry operator to proceed.

machine-config is failing as one of master node is spewing errors like:
```console
I0320 19:04:42.963261    8911 daemon.go:638] In bootstrap mode
I0320 19:04:42.963267    8911 daemon.go:666] Current+desired config: master-66e93c61606a6f5ed4b7ccaacc53b15e
I0320 19:04:42.966939    8911 daemon.go:436] Detected a login session before the daemon took over on first boot
I0320 19:04:42.966948    8911 daemon.go:437] Applying annotation: machineconfiguration.openshift.io/ssh
E0320 19:04:43.142669    8911 writer.go:119] Marking Degraded due to: parsing booted osImageURL: parsing reference: "<not pivoted>": invalid reference format
W0320 19:04:43.342369    8911 daemon.go:292] Booting the MCD errored with parsing booted osImageURL: parsing reference: "<not pivoted>": invalid reference format
I0320 19:04:43.342384    8911 run.go:22] Running captured: rpm-ostree status
I0320 19:04:43.363156    8911 daemon.go:738] State: idle
AutomaticUpdates: disabled
Deployments:
* ostree://rhcos:tmpref-rhcos
                   Version: 410.8.20190319.1 (2019-03-19T18:14:26Z)
                    Commit: 1f774246c0b1d2a5edec789634f2e3ea5e2168534cae68a97a3172b971608349
I0320 19:04:43.363179    8911 daemon.go:638] In bootstrap mode
I0320 19:04:43.363186    8911 daemon.go:666] Current+desired config: master-66e93c61606a6f5ed4b7ccaacc53b15e
I0320 19:04:43.366834    8911 daemon.go:436] Detected a login session before the daemon took over on first boot
I0320 19:04:43.366847    8911 daemon.go:437] Applying annotation: machineconfiguration.openshift.io/ssh
```

abhinavdahiya commented:
pushed https://github.com/openshift/installer/compare/652b5fc2f205a02b29b798809c16352a8b939ad9..7dee4734565059deb629a264093355ac92c3ef30#diff-1a64e7030769df4d25b1c09ddb8f62de to document how to unstick registry by setting storage backend to `emptyDir`

abhinavdahiya commented:
So got a cluster up with only one clusteroperator failing :fireworks: 
```
oc --config dev-metal/auth/kubeconfig get co -ojson | jq '.items[] | select(.status.conditions[] | .type == "Available" and .status != "True") | [.metadata.name, .status.conditions]'
[
  "machine-config",
  [
    {
      "lastTransitionTime": "2019-03-21T16:57:28Z",
      "message": "Cluster not available for 4.0.0-0.ci-2019-03-21-143835",
      "status": "False",
      "type": "Available"
    },
    {
      "lastTransitionTime": "2019-03-21T16:57:28Z",
      "message": "Cluster is bootstrapping 4.0.0-0.ci-2019-03-21-143835",
      "status": "True",
      "type": "Progressing"
    },
    {
      "lastTransitionTime": "2019-03-21T16:58:06Z",
      "message": "Failed to resync 4.0.0-0.ci-2019-03-21-143835 because: error pool master is not ready. status: (total: 1, updated: 0, unavailable: 0)",
      "reason": "error pool master is not ready. status: (total: 1, updated: 0, unavailable: 0)",
      "status": "True",
      "type": "Failing"
    }
  ]
]
```
issue: https://gitlab.cee.redhat.com/coreos/redhat-coreos/issues/28.

abhinavdahiya commented:
Okay so i tried to run some tests:
```sh
KUBECONFIG=/home/adahiya/go/src/github.com/openshift/installer/upi/metal/dev-metal/auth/kubeconfig _output/local/bin/linux/amd64/openshift-tests run openshift/conformance/parallel
```
[results]
```plain
The bootstrap user should successfully login with password decoded from kubeadmin secret [Suite:openshift/conformance/parallel]
[Area:Networking] network isolation when using a plugin that does not isolate namespaces by default should allow communication between pods in different namespaces on the same node [Suite:openshift/conformance/parallel]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should expose a health check on the metrics port [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should expose the profiling endpoints [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should respond with 503 to unrecognized hosts [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should serve routes that were created from an ingress [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should set Forwarded headers appropriately [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should support reencrypt to services backed by a serving certificate automatically [Suite:openshift/conformance/parallel/minimal]
[Conformance][templates] templateinstance readiness test  should report ready soon after all annotated objects are ready [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution custom [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution docker [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution s2i [Suite:openshift/conformance/parallel]
[Feature:Builds][Conformance] oc new-app  should fail with a --name longer than 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in docker build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in sti build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in docker build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in s2i build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  [Conformance] buildconfigs should have a default history limit set when created via the group api [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune builds after a buildConfig change [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune canceled builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune completed builds based on the successfulBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune errored builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune failed builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][timing] capture build stages and durations  should record build stages and durations for s2i [Suite:openshift/conformance/parallel]
[Feature:DeploymentConfig] deploymentconfigs keep the deployer pod invariant valid [Conformance] should deal with cancellation of running deployment [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs paused [Conformance] should disable actions on deployments [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs rolled back [Conformance] should rollback to an older deployment [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs should respect image stream tag reference policy [Conformance] resolve the image pull spec [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs when run iteratively [Conformance] should only deploy the last deployment [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with failing hook [Conformance] should get all logs from retried hooks [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with a trigger used by different containers [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with multiple triggers [Suite:openshift/conformance/parallel/minimal]
[Feature:Platform] Managed cluster should have no crashlooping pods in core namespaces over two minutes [Suite:openshift/conformance/parallel]
[Feature:Platform][Smoke] Managed cluster should start all core operators [Suite:openshift/conformance/parallel]
[image_ecosystem][mongodb] openshift mongodb image  creating from a template should instantiate the template [Suite:openshift/conformance/parallel]
[k8s.io] Container Lifecycle Hook when create a pod with lifecycle hook should execute prestop http hook properly [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[k8s.io] Pods should support pod readiness gates [NodeFeature:PodReadinessGate] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD with validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD without validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] ReplicationController should adopt matching pods on creation [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should adopt matching orphans and release non-matching pods [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should perform rolling updates and roll backs of template modifications with PVCs [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-auth] [Feature:NodeAuthorizer] Getting a non-existent configmap should exit with the Forbidden error, not a NotFound error [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Kubectl patch should add annotations for pods in rc  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Kubectl run --rm job should create a job from an image, then delete the job  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Kubectl run job should create a job from an image when restart is OnFailure  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-network] Networking Granular Checks: Pods should function for node-pod communication: http [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-network] Services should be able to change the type from ExternalName to ClusterIP [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-network] Services should create endpoints for unready pods [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-network] Services should have session affinity work for service with type clusterIP [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] ResourceQuota should create a ResourceQuota and capture the life of a pod. [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] Ephemeralstorage When pod refers to non-existent ephemeral storage should allow deletion of pod with invalid volume : configmap [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] Ephemeralstorage When pod refers to non-existent ephemeral storage should allow deletion of pod with invalid volume : projected [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] GCP Volumes NFSv4 should be mountable for NFSv4 [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: azure] [Testpattern: Inline-volume (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: azure] [Testpattern: Inline-volume (default fs)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: cinder] [Testpattern: Dynamic PV (default fs)] subPath should fail for new directories when readOnly specified in the volumeSource [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: cinder] [Testpattern: Dynamic PV (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Pre-provisioned PV (default fs)] subPath should support file as subpath [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (ext3)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Inline-volume (default fs)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: vSphere] [Testpattern: Pre-provisioned PV (ext4)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify "immediate" deletion of a PVC that is not in active use by a pod [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify that PVC in active use by a pod is not removed immediately [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify that scheduling of a pod that uses PVC that is being deleted fails and the pod becomes Unschedulable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PersistentVolumes NFS with Single PV - PVC pairs create a PVC and a pre-bound PV: test write access [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PersistentVolumes NFS with Single PV - PVC pairs create a PVC and non-pre-bound PV: test write access [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] Secrets should be consumable from pods in volume [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
```

abhinavdahiya commented:
[test run 2] `error: 39 fail, 482 pass, 382 skip (42m18s)`
<Details>

```
Failing tests:

The bootstrap user should successfully login with password decoded from kubeadmin secret [Suite:openshift/conformance/parallel]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should expose a health check on the metrics port [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should expose the profiling endpoints [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should respond with 503 to unrecognized hosts [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should serve routes that were created from an ingress [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should set Forwarded headers appropriately [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should support reencrypt to services backed by a serving certificate automatically [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution custom [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution docker [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution s2i [Suite:openshift/conformance/parallel]
[Feature:Builds][Conformance] oc new-app  should fail with a --name longer than 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in docker build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in sti build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in docker build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in s2i build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  [Conformance] buildconfigs should have a default history limit set when created via the group api [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune builds after a buildConfig change [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune canceled builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune completed builds based on the successfulBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune errored builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune failed builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:DeploymentConfig] deploymentconfigs with failing hook [Conformance] should get all logs from retried hooks [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with a trigger used by different containers [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with multiple triggers [Suite:openshift/conformance/parallel/minimal]
[Feature:Platform] Managed cluster should have no crashlooping pods in core namespaces over two minutes [Suite:openshift/conformance/parallel]
[Feature:Platform][Smoke] Managed cluster should start all core operators [Suite:openshift/conformance/parallel]
[image_ecosystem][mongodb] openshift mongodb image  creating from a template should instantiate the template [Suite:openshift/conformance/parallel]
[k8s.io] Pods should support pod readiness gates [NodeFeature:PodReadinessGate] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD with validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD without validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should adopt matching orphans and release non-matching pods [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should perform rolling updates and roll backs of template modifications with PVCs [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify "immediate" deletion of a PVC that is not in active use by a pod [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify that PVC in active use by a pod is not removed immediately [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PVC Protection Verify that scheduling of a pod that uses PVC that is being deleted fails and the pod becomes Unschedulable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] PersistentVolumes NFS with Single PV - PVC pairs create a PVC and non-pre-bound PV: test write access [Suite:openshift/conformance/parallel] [Suite:k8s]

Writing JUnit report to junit-bare-metal-upi/0/junit_e2e_20190321-215132.xml

error: 39 fail, 482 pass, 382 skip (42m18s)
```
[junit_e2e_20190321-215132.xml.txt](https://github.com/openshift/installer/files/2994109/junit_e2e_20190321-215132.xml.txt)

abhinavdahiya commented:
> Conformance][Area:Networking][Feature:Router] The HAProxy router should expose a health check on the metrics port [Suite:openshift/conformance/parallel/minimal]
> [Conformance][Area:Networking][Feature:Router] The HAProxy router should expose the profiling endpoints [Suite:openshift/conformance/parallel/minimal]
> [Conformance][Area:Networking][Feature:Router] The HAProxy router should respond with 503 to unrecognized hosts [Suite:openshift/conformance/parallel/minimal]
> [Conformance][Area:Networking][Feature:Router] The HAProxy router should serve routes that were created from an ingress [Suite:openshift/conformance/parallel/minimal]
> [Conformance][Area:Networking][Feature:Router] The HAProxy router should set Forwarded headers appropriately [Suite:openshift/conformance/parallel/minimal]
> [Conformance][Area:Networking][Feature:Router] The HAProxy router should support reencrypt to services backed by a serving certificate automatically [Suite:openshift/conformance/parallel/minimal]
> [F

sdodson commented:
/retest
/lgtm

wking commented:
[e2e-aws][1] was:

```
Failing tests:

[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Inline-volume (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Pre-provisioned PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
```
openshift/origin#22412

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1416/pull-ci-openshift-installer-master-e2e-aws/4672

bertinatto commented:
/retest

wking commented:
[e2e-aws][1]:

```
level=fatal msg="failed to initialize the cluster: Working towards 0.0.1-2019-03-27-122957: 99% complete: timed out waiting for the condition"
```

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1416/pull-ci-openshift-installer-master-e2e-aws/4686

wking commented on docs/user/metal/install_upi.md:
```diff
@@ -0,0 +1,326 @@
+
+    As the etcd members are located on the control plane machines. Each control plane machine requires connectivity to [etcd server][etcd-ports], [etcd peer][etcd-ports] and [etcd-metrics][etcd-ports] on every other control plane machine.
+
+    All the worker machines should have connectivity to [etcd server][etcd-ports] and [etcd-metrics][etcd-ports] ports on the control plane machines.
```
@abhinavdahiya, do you remember why compute machines need access to etcd?  I expect they need etcd-metrics in case they end up running Prometheus (is this #1334), but I don't understand why compute nodes would need direct access to the etcd server itself.  CC @katherinedube

wking replied:
We don't need this for IPI AWS, where [we only allow master->master etcd][1].

[1]: https://github.com/openshift/installer/blob/d1cb735dfe24c664463278ed995e10e342024b0d/data/data/aws/vpc/sg-master.tf#L186

wking replied:
From @abhinavdahiya, at some point folks were scraping etcd metrics from the server port.  We can remove this once we here from etcd/monitoring that they no longer need to scrape metrics from the server port.

wking replied:
Evidence of etcd-metrics getting scraped from their own port:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/706/artifacts/e2e-aws-upgrade/pods.json | grep -o 'Delete endpoint [0-9:.]* for service ["\\]*openshift-etcd/etcd:etcd-metrics' | sort | uniq
Delete endpoint 10.0.149.41:9979 for service \"openshift-etcd/etcd:etcd-metrics
```

Is that enough for us to pull the plug on recommending this server-port access?

wking replied:
Dropping in #1671
